test/e2e/e2e_test.go

```diff
@@ -0,0 +1,445 @@
+package e2e
+
+import (
+	"context"
+	"io"
+	"io/ioutil"
+	"os"
+	"os/user"
+	"path/filepath"
+	"strconv"
+	"strings"
+	"sync"
+	"testing"
+	"time"
+
+	"github.com/submariner-io/armada/pkg/deploy"
+	"github.com/submariner-io/armada/pkg/image"
+	"github.com/submariner-io/armada/pkg/wait"
+	"sigs.k8s.io/kind/pkg/cluster/nodes"
+
+	dockertypes "github.com/docker/docker/api/types"
+	"github.com/docker/docker/api/types/filters"
+	dockerclient "github.com/docker/docker/client"
+	"github.com/gobuffalo/packr/v2"
+	log "github.com/sirupsen/logrus"
+	createclustercmd "github.com/submariner-io/armada/cmd/armada/create/cluster"
```

Comment:
I'll put a couple of comments on the structure of the project here since you have the majority of the imports here but applies in general..

1. RE: `armada/cmd/armada/create/cluster` and `armada/cmd/armada/destroy/cluster`
    * A general comment, why repeating `armada` twice (before/after `cmd`)?
    * Why `destroy` and `create` are they in different modules? It makes more sense to have them in a `cluster` module.

2. RE: `armada/pkg/cluster` and `armada/pkg/defaults`
    * Why are they in different modules separate from the `cmd` hierarchy?

3. In general the most of the modules within `cmd` should not there. For example, the `cluster` module should exist independent from `cmd` and should provide the `create` and `deploy` functions and `cmd` then should have all the wrapping to call those functions in the context of a CLI.

4. RE: too specific commands:
    * the `export/logs` command should be an `execute` command instead that passes that execution to each `kind` separately (since anyway internally you are doing just that but calling just one specific `kind` command)
    * the `deploy/nginx` command should instead be an `apply` that is passed to `kind` and thus can be generalized. As a corollary of this, the YAMLs specific to doing these deployments should not be here.

Reply:
> 1. RE: `armada/cmd/armada/create/cluster` and `armada/cmd/armada/destroy/cluster`
>    
>    * A general comment, why repeating `armada` twice (before/after `cmd`)?
>    * Why `destroy` and `create` are they in different modules? It makes more sense to have them in a `cluster` module.
> 3. In general the most of the modules within `cmd` should not there. For example, the `cluster` module should exist independent from `cmd` and should provide the `create` and `deploy` functions and `cmd` then should have all the wrapping to call those functions in the context of a CLI.

Done with https://github.com/submariner-io/armada/pull/4